packages/ui-react/src/components/Payment/Payment.tsx

```diff
@@ -223,24 +160,15 @@ const Payment = ({
                   <Button
                     className={styles.upgradeSubscription}
                     label={t('user:payment.change_subscription')}
-                    disabled={!canRenewSubscription && activeSubscription.status === 'cancelled'}
-                    onClick={() => {
-                      if (offers.length > 1 && !canRenewSubscription) {
-                        setIsChangingOffer(true);
-                      } else {
-                        onUpgradeSubscriptionClick?.();
-                      }
-                    }}
+                    disabled={(!canRenewSubscription && activeSubscription.status === 'cancelled') || !!pendingOffer}
+                    onClick={onUpgradeSubscriptionClick}
                     fullWidth={isMobile}
                     color="primary"
                     data-testid="change-subscription-button"
                   />
                 )
               )}
-              {(activeSubscription.status === 'active' || activeSubscription.status === 'active_trial') &&
-              !isGrantedSubscription &&
-              !isChangingOffer &&
-              canRenewSubscription ? (
+              {(activeSubscription.status === 'active' || activeSubscription.status === 'active_trial') && !isGrantedSubscription && canRenewSubscription ? (
```

Comment:
This condition should be checked. I think the `canRenewSubscription` should be removed for the cancel subscription button. So:

1. has subscription and is active -> Show "Cancel Subscription" button
2. has subscription and is not active and canRenewSubscription -> show "Renew Subscription" button